sdk/search/test-resources.json

```diff
@@ -1,8 +1,111 @@
+          }
+      }
+  },
+  "variables": {
+      "searchServiceName": "[concat('azs-java-', parameters('baseName'))]",
+      "searchApiVersion": "2020-03-13",
+      "storageAccountName": "[concat('azsstg', parameters('baseName'))]",
+      "storageContainerName": "[concat('azscontainer', parameters('baseName'))]",
+      "storageApiVersion": "2019-06-01"
+  },
+  "resources": [
+      {
+          "name": "[variables('searchServiceName')]",
+          "type": "Microsoft.Search/searchServices",
+          "apiVersion": "[variables('searchApiVersion')]",
+          "location": "[parameters('location')]",
+          "sku": {
+              "name": "[parameters('searchSku')]"
+          },
+          "properties": {
+              "replicaCount": 1,
+              "partitionCount": 1,
+              "hostingMode": "default",
+              "publicNetworkAccess": "Enabled",
+              "networkRuleSet": {
+                  "ipRules": []
+              }
+          }
+      },
+      {
+          "name": "[variables('storageAccountName')]",
+          "type": "Microsoft.Storage/storageAccounts",
+          "apiVersion": "[variables('storageApiVersion')]",
+          "location": "[parameters('location')]",
+          "properties": {
+              "accessTier": "Hot",
+              "supportsHttpsTrafficOnly": true
+          },
+          "sku": {
+              "name": "Standard_LRS",
+              "tier": "Standard"
+          },
+          "kind": "StorageV2",
+          "resources": [
+            {
+              "type": "blobServices/containers",
+              "name": "[concat('default/', variables('storageContainerName'))]",
+              "apiVersion": "[variables('storageApiVersion')]",
+              "properties": {},
+              "dependsOn": [
+                "[variables('storageAccountName')]"
+              ]
+            }
+          ]
+      }
+  ],
+  "outputs": {
+      "AZURE_SEARCH_SERVICE_ENDPOINT": {
+          "type": "string",
+          "value": "[concat('https://', variables('searchServiceName'), '.search.windows.net')]"
+      },
+      "AZURE_SEARCH_SERVICE_API_KEY": {
+          "type": "string",
+          "value": "[listAdminKeys(variables('searchServiceName'), variables('searchApiVersion')).primaryKey]"
+      },
+      "AZURE_SEARCH_QUERY_API_KEY": {
+          "type": "string",
+          "value": "[listQueryKeys(variables('searchServiceName'), variables('searchApiVersion')).value[0].key]"
+      },
+      "AZURE_SEARCH_STORAGE_NAME": {
+          "type": "string",
+          "value": "[variables('storageAccountName')]"
+      },
+      "AZURE_SEARCH_STORAGE_CONNECTION_STRING": {
+        "type": "string",
+        "value": "[concat('DefaultEndpointsProtocol=https;AccountName=', variables('storageAccountName'), ';AccountKey=', listKeys(variables('storageAccountName'), variables('storageApiVersion')).keys[0].value, ';EndpointSuffix=core.windows.net')]"
```

Comment:
Guidance from the Engineering team is not include hardcoded endpoint suffixes but keep them configurable esp for storage since they do not resolve correctly in Azure public clouds.
More info - https://github.com/Azure/azure-sdk-for-java/issues/10380#issuecomment-617424971

Reply:
Will be added in the next commit.